src/main/java/org/mockito/internal/creation/instance/ConstructorInstantiator.java

```diff
@@ -5,26 +5,30 @@
 package org.mockito.internal.creation.instance;
 
 import java.lang.reflect.Constructor;
+import java.util.Arrays;
+
 import org.mockito.internal.util.reflection.AccessibilityChanger;
 
 import static org.mockito.internal.util.StringUtil.join;
 
 public class ConstructorInstantiator implements Instantiator {
 
-    private final Object outerClassInstance;
+    private final Object[] constructorArgs;
+
+    public ConstructorInstantiator(Object... constructorArgs) {
+        if (constructorArgs != null) {
+            this.constructorArgs = constructorArgs;
+        } else {
+            this.constructorArgs = new Object[0];
```

Comment:
It should not be possible  to pass a null vararg-array here, as this probably indicates an bug upper in the call hierarchy. We should better check the precondition here instead of correcting the input. WDYT?